tests/common/output/test_sarif_report.py

```diff
@@ -346,8 +350,97 @@ def test_multiple_instances_of_same_rule_do_not_break_schema(self):
         self.assertTrue(are_rule_indexes_correct_in_results(sarif.json))
         self.assertTrue(are_rules_without_help_uri_correct(sarif.json))
 
+    def test_non_url_guideline_link(self):
```

Comment:
maybe better move the resources here to a fixture so the test will be more readable

Reply:
unittest doesn't have `pytest` style fixture, but I moved out to a simple function